### PR TITLE
Unsafe Serialization for Poseidon Constants

### PR DESCRIPTION
**Pull request by winston-h-zhang**
This PR implements the `Abomonation` traits necessary for the upstream of faster public parameter caching in Nova. See https://github.com/lurk-lab/lurk-rs/pull/399 for details.

(This is also a copy-paste of https://github.com/winston-h-zhang/neptune/pull/1, moving the work there to a `lurk-lab` fork.)

### CODE REVIEW

**Review by huitseeker — COMMENTED** [review]
This needs https://github.com/lurk-lab/bellperson/commit/71930b5b5e98763a3337e61365c213b23d7d79f6 i.e. https://github.com/lurk-lab/bellperson/pull/3 for CI to pass, otherwise LGTM.

File: Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,8 @@ log = "0.4.17"
 pasta_curves = { git="https://github.com/lurk-lab/pasta_curves", branch="dev", features = ["serde"] }
 trait-set = "0.3.0"
 serde = { version = "1.0", features = ["derive"] }
+abomonation = "0.7.3"
+abomonation_derive = { git = "https://github.com/winston-h-zhang/abomonation_derive.git" }
```
Comment (huitseeker) [inline Cargo.toml:27]: Do you have a PR opened upstream to extend this, since IIUC it's a strict improvement over the proc-macro therein?

**Review by huitseeker — CHANGES_REQUESTED** [review]
This needs a rebase!

File: Cargo.toml (hunk 1 above)
Comment (winston-h-zhang) [reply Cargo.toml:27]: I don't think `abomonation_derive` is maintained anymore. I could open a PR, but we will have to withhold the expectation of it getting anywhere

Comment (huitseeker) [reply Cargo.toml:27]: Fine by me: the reason we do all this is poor expectations.

**Review by huitseeker — APPROVED** [review]
Thanks a bunch!

**Pull request merged**